### DISCUSSION TITLE
Could not infer the number of file handles used

### DISCUSSION THREAD

**Discussion by NickCc6** [rabbitmq/rabbitmq-server#7977] [Other]
### Describe the bug

On my Windows server, it often stops working and I don't know how to handle this issue
I can see from the log: Could not infer the number of file handles used
Can you help me?
![image](https://user-images.githubusercontent.com/42535452/234193616-b56d43c6-1afa-4955-9864-d13c1f0a94e2.png)
[logfile](https://github.com/NickCc6/NickCc6/blob/main/rabbit%40ecs-1d22.log)

### Reproduction steps

1.install  Erlang 25.3、 RabbitMQ 3.11.13、rabbitmq_delayed_message_exchange-3.11.1.ez
2.Add a new vhost and Add a delayed_message
3.After a random period of time, the service will stop

### Expected behavior

normal operation

**Comment by michaelklishin** [comment]
Please never use GitHub issues for questions if the repository has Discussions enabled.

```
2023-04-25 14:02:24.612000+08:00 [info] <0.95.0> RabbitMQ is asked to stop...
```

suggests that the node was told to stop using `rabbitmqctl.bat shutdown` or similar.

I don't see how this is related to the number of file handles. If your Windows system does not have
[`handle.exe`](https://learn.microsoft.com/en-us/sysinternals/downloads/handle) in `PATH`, nodes will not be able to retrieve their file handle use but that does not prevent them from running. Only
one (fairly important) metric will not be available, that's it.

**Reply by NickCc6** [reply]
How should I investigate this problem? I'm at a loss right now.
The service often stops, and I cannot obtain answers on how to handle it from the logs
What I see in the system log is: RabbitMQ: Erlang machine seams to die continuously, not restarted
Do you have any good suggestions？

**Reply by michaelklishin** [reply]
Use system logs of some kind, RabbitMQ cannot know what invoked the CLI command.

**Comment by lukebakken** [comment]
> What I see in the system log is: RabbitMQ: Erlang machine seams to die continuously, not restarted

You probably have antivirus software or some other security scanner killing the `erl.exe` process. Start by **disabling** all of that.

**Reply by NickCc6** [reply]
Okay, I'll give it a try. Thank you

**Comment by NickCc6** [comment]
I couldn't find the cause of the problem, but after restarting the server, everything was fine. I will continue to observe in the future

**Comment by NickCc6** [comment]
@lukebakken   @michaelklishin 
Sorry to bother you. After stabilizing for three weeks, this situation has occurred again. This is the Windows system log. Can you identify the cause of the problem or do you have any good troubleshooting suggestions

![image](https://github.com/rabbitmq/rabbitmq-server/assets/42535452/d54d0d73-0d9b-4259-8ce2-0e7422c30df9)

[logfile](https://github.com/NickCc6/NickCc6/blob/main/ErlSrvlog.txt)
[RabbitMQlogfile](https://github.com/NickCc6/NickCc6/blob/main/2023519rabbit%40ecs-1d22.log)

**Reply by lukebakken** [reply]
That's very strange. The RabbitMQ log files only show that RabbitMQ starts up after an abrupt halt (you don't see any "Shutting down..." messages).

What other software is running on these servers? You should _only_ be running Windows (of course) and RabbitMQ. *Nothing else* - no security scanners, no antivirus, no Windows Defender, no indexing service, etc. Shut everything else off and see if the problem continues.

Also, are you monitoring these systems so you could correlate some system stats to these events?

See if any of these apply to your environment. I searched google with the string `erlang machine seems to die continuously`:

* MSMQ is installed - https://stackoverflow.com/questions/53829827/rabbitmq-service-starts-and-then-stopped-due-to-rabbitmq-erlang-machine-seems-t#comment94509726_53829827
* Multiple Erlangs are installed - https://github.com/rabbitmq/rabbitmq-server/issues/1969
* https://www.google.com/search?q=%22erlang+machine+seems+to+die+continously+not+restarted

**Reply by NickCc6** [reply]
Thank you. I'll try to handle it this way

**Reply by codecool1998** [reply]
Hello, I also encountered the same problem, may I ask you to find the solution to the problem

**Reply by michaelklishin** [reply]
@codecool1998 you are not entitled to a "solution to the problem", either you investigate it and report back, or pay someone to do that for you. Open source does not mean "free labor you can ask to do things for you on GitHub."
